src/test/java/bio/terra/common/FutureUtilsTest.java

```diff
@@ -139,19 +134,57 @@ public void testWaitForTaskAcrossTwoBatches() throws InterruptedException {
     assertThat(CollectionUtils.collect(batch1Futures, Future::isDone)).doesNotContain(true);
 
     final List<Integer> resolved1 = FutureUtils.waitFor(batch1Futures);
-    assertThat(resolved1).containsAll(IntStream.range(0, 3).boxed().collect(Collectors.toList()));
+    assertThat(resolved1).containsAll(IntStream.range(0, 3).boxed().toList());
     assertThat(executorService.getActiveCount()).isZero();
   }
 
   @Test
-  public void testWaitForTasksSomeFail() throws InterruptedException {
+  void testWaitForTasksSomeFail() {
```

Comment:
More tests here seems fine although I don't think you need to actually use threads or async operations to test the functionality in `FutureUtils.waitFor()`, since you can create a `Future` directly here without creating it in a thread.

Reply:
I'm glad you brought this up -- I was thinking that we were doing more than we needed to do to test this functionality!

Reply:
I really like the resulting simplification here, rather than testing executor functionality and putting sleeps in new tests (which were flagged as code smells).